### ISSUE TITLE
[Accessibility] Audio Cues doesn't work in web editor

### ISSUE THREAD

**Issue by jooyoungseo** [microsoft/vscode#147190]
CC @isidorn:

Version: 1.66.0-insider
Commit: 521a3d858570ebc9ce9d5602802f4525a02e045c
Date: 2022-03-29T07:32:43.053Z
Browser: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/100.0.4896.75 Safari/537.36

## Reproducible steps

1. Run `Help: List Audio Cues` and see if you hear any audio cues.
2. It does not work.

**Comment by Trandung5698** [comment]
Vào 08:50, T.2, 11 Th4, 2022 JooYoung Seo ***@***.***> đã
viết:

**Comment by hediet** [comment]
We forgot to test that feature indeed for vscode web.
That's the error:

> Refused to load media from 'https://vscodeweb.azureedge.net/insider/521a3d858570ebc9ce9d5602802f4525a02e045c/out/vs/workbench/contrib/audioCues/browser/media/error.opus' because it violates the following Content Security Policy directive: "media-src 'none'".

**Comment by jooyoungseo** [comment]
Has this issue been resolved? I still don't hear audio cues in the latest GitHub VSCode insider.

**Comment by isidorn** [comment]
I just tried with https://insiders.vscode.dev/ which contains the fix and I can still reproduce the problem. Folded areas do not produce a sound for me. Let me reopen and @hediet can also double check on his machine.

**Comment by hediet** [comment]
Hmm, it actually works on my machine. @isidorn can you clear all caches and try again?

**Comment by isidorn** [comment]
@hediet I just cleared caches and tried in a new incognito window and still does not work - weird. My steps
1. insiders.vscode.dev 
2. Open folder locally -> choose vscode
3. Set that error audio cues get read
4. Open cglinceses.json top level, there are a bunch of errors and none produce a sound when navigating on that line

I might be doing something wrong.

@jooyoungseo can you repro with insiders.vscode.dev?

**Comment by hediet** [comment]
@isidorn can you try out the help command for audio cues?

**Comment by isidorn** [comment]
@hediet yup I can reproduce with the help command. Here's an error from the dev console
```
Error while playing sound DOMException: Failed to load because no supported source was found.
playSound @ audioCueService.ts:77
```

**Comment by hediet** [comment]
After updating all my browsers, it works for me in the latest chrome, firefox and edge.
However, [safari does not seem to support opus](https://caniuse.com/opus). Maybe we should provide a mp3 file for unsupported browsers?

**Comment by isidorn** [comment]
@deepak1556 can we control what audio files work for electron?
Otherwise we would have to ship different audio files based on if we are in electron or web.

**Comment by deepak1556** [comment]
Current builds only supports the following non-proprietary codecs

* .opus
* .webm (encoded with opus codec, the default encoding with vorbis codec is not supported)
* .flac

We can get support for proprietary codecs, but we would have to go through some CELA loop before it can be shipped. It has been a while so I don't know the exact steps. I can ask around MS-Teams runtime folks who might be familiar with shipping proprietary codecs.

**Comment by isidorn** [comment]
Assigning to August since we will be able to use mp3 soon.
@hediet I have reached out to the sound designer so he provides all the audio cues in mp3.

**Comment by isidorn** [comment]
We will no longer ship the opus codec in the next release. So the change to mp3 is a must do :)
Our sound designer confirmed he will provide sounds in mp3 next week.

**Comment by isidorn** [comment]
Deepak has updated electron so this should be possible now. 
And the sound designer has provided MP3 files for these sounds.

**Comment by jooyoungseo** [comment]
@isidorn -- Thanks for the update, but the Audio Cues does not work in github.dev. It's working in vscode.dev though. Does it take some time for github.dev?

**Comment by isidorn** [comment]
@jooyoungseo yeah, github.dev updates a bit slower than vscode.dev. I expect that they will pick up the update in the next couple of weeks.
Glad to hear you are using vscode.dev. Do let us know if you have any other feedback regarding it. Thanks!

**Comment by jooyoungseo** [comment]
@isidorn -- Aw, I see. Thanks for your prompt reply. I will keep checking github.dev update, and will keep you posted.

**Comment by jooyoungseo** [comment]
@isidorn -- Please correct me if I am wrong, but github.dev still does not support the audio cues while it seems to have picked up the update.

**Comment by jooyoungseo** [comment]
Aw, I see. @isidorn, thanks for investigating this. I will wait some more weeks then.

**Comment by meganrogge** [comment]
Seeing the same error as above

**Comment by isidorn** [comment]
fyi @joaomoreno since I think the fixed it the last time and might have more context.

**Comment by joaomoreno** [comment]
@bpasero Can you check if the recent changes to CSP caused this to regress?

**Comment by bpasero** [comment]
Oh 🤦 , yeah its the removal of `https:`.

**Comment by bpasero** [comment]
PR up: https://github.com/microsoft/vscode-dev/pull/879

**Comment by jrieken** [comment]
works in safari

**Comment by meganrogge** [comment]
@Jooyoungseo has reported this still doesn't work on Chrome, which is an important issue

**Comment by isidorn** [comment]
I just tried and audio cues work for me both in Chrome and Edge. Here are my steps:
1. Open vscode.dev
2. F1 > list audio cues
3. Navigate through them -> hear sound ✅ 

@meganrogge can you reproduce the issue?

**Comment by meganrogge** [comment]
Able to reproduce in `github.dev` with the error message:
```
5Refused to load media from '<URL>' because it violates the following Content Security Policy directive: "media-src <URL>
            <URL>".
```
![Screenshot 2023-07-24 at 11 05 41 AM](https://github.com/microsoft/vscode/assets/29464607/595c59fe-b73f-4894-89e7-5498eee6d8e2)

**Comment by bpasero** [comment]
@meganrogge that is not under our control though, GitHub controls the content security policy, they will have to do a similar change as we did: https://github.com/microsoft/vscode-dev/pull/879/files

**Comment by isidorn** [comment]
@bpasero is there a github.dev repo where we can go and file an issue for this, so we mark this as upstream?

**Comment by bpasero** [comment]
@isidorn I would talk to @jkeech or Gideon Cheruiyot.

All I know is the Codespaces repo: https://github.com/github/codespaces

**Comment by isidorn** [comment]
Thanks. Since the issue repros also in a Codespace I opened an issue here https://github.com/github/codespaces/issues/14609

Thus marking this as an upstream and moving to backlog since we can not fix it on our side.
